checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInferenceLogic.java

```diff
@@ -52,95 +94,621 @@ public class MustCallInferenceLogic {
+    if (cmValue != null) {
+      for (AnnotationMirror anno : cmValue.getAnnotations()) {
+        if (AnnotationUtils.areSameByName(
+            anno, "org.checkerframework.checker.calledmethods.qual.CalledMethods")) {
+          cmAnno = anno;
+          break;
+        }
+      }
+    }
+
+    if (cmAnno == null) {
+      cmAnno = typeFactory.top;
+    }
+
+    AnnotationMirror cmAnnoForMustCallMethods =
+        typeFactory.createCalledMethods(mustCallValues.toArray(new String[0]));
+    if (!mustCallValues.isEmpty()
+        && typeFactory.getQualifierHierarchy().isSubtype(cmAnno, cmAnnoForMustCallMethods)) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Updates the set of obligations any must-call-alias parameters passed to the given method
+   * invocation or object creation node.
+   *
+   * @param obligations the set of obligations to update
+   * @param node the node representing the method invocation or object creation
+   */
+  private void updateMethodInvocationOrObjectCreationNode(Set<Obligation> obligations, Node node) {
+    List<Node> arguments = mcca.getArgumentsOfInvocation(node);
+    List<? extends VariableElement> parameters = mcca.getParametersOfInvocation(node);
+
+    for (int i = 0; i < arguments.size(); i++) {
+      Node tempVar = mcca.getTempVarOrNode(arguments.get(i));
+      if (!(tempVar instanceof LocalVariableNode)) {
+        continue;
+      }
+
+      Obligation obligation =
+          MustCallConsistencyAnalyzer.getObligationForVar(obligations, (LocalVariableNode) tempVar);
+      if (obligation == null) {
+        continue;
+      }
+
+      Node method = mcca.getTempVarOrNode(node);
+      if (typeFactory.hasMustCallAlias(parameters.get(i))) {
+        Set<ResourceAlias> newResourceAliasesForObligation =
+            new LinkedHashSet<>(obligation.resourceAliases);
+        newResourceAliasesForObligation.add(
+            new ResourceAlias(new LocalVariable((LocalVariableNode) method), node.getTree()));
+        obligations.remove(obligation);
+        obligations.add(new Obligation(newResourceAliasesForObligation));
       }
     }
   }
 
   /**
    * Updates {@code worklist} with the next block along all paths to the regular exit point. If the
    * next block is a regular exit point, adds an {@literal @}Owning annotation for fields in {@link
-   * #owningFields}.
+   * #owningFieldToECM}.
    *
    * @param curBlock the current block
    * @param visited set of blocks already on the worklist
    * @param worklist current worklist
    */
-  private void propagateRegPaths(Block curBlock, Set<Block> visited, Deque<Block> worklist) {
+  private void propagateRegPaths(
+      Set<Obligation> obligations,
+      Block curBlock,
+      Set<BlockWithObligations> visited,
+      Deque<BlockWithObligations> worklist) {
 
     List<Block> successors = getNormalSuccessors(curBlock);
 
-    for (Block b : successors) {
+    for (Block successor : successors) {
```

Comment:
This is a fine change.  However, it does clutter the pull request a bit.  If you could make a pull request with cleanups such as these, then merge it into master, then this diff won't show up in this pull request any longer.

Reply:
Thanks for the suggestion. I will create a new PR for these cleanups.